meteor/client/ui/Settings/ShowStyle/VariantListItem.tsx

```diff
@@ -95,6 +95,80 @@ export const VariantListItem: React.FunctionComponent<IShowStyleVariantItemProps
 		})
 	}
 
+	function showStyleVariantNameAlreadyExists(): boolean {
+		return props.dragDropVariants.some((variant: ShowStyleVariant) => {
+			return variant.name === props.showStyleVariant.name && variant._id !== props.showStyleVariant._id
+		})
+	}
+
+	function addDuplicationCountToName(showStyleVariant: ShowStyleVariant, duplicatorIncrement?: number): void {
+		const increment = duplicatorIncrement ?? 0
+		showStyleVariant.name = getNameWithRemovedDuplicateIndicator(showStyleVariant)
+		showStyleVariant.name += ' (' + (getDuplicatedShowStyleVariantCount(showStyleVariant) + increment) + ')'
+		if (showStyleVariantNameAlreadyExists()) {
+			addDuplicationCountToName(showStyleVariant, increment + 1)
+		} else {
+			MeteorCall.showstyles.renameShowStyleVariant(props.showStyleBase._id, props.showStyleVariant).catch(logger.warn)
+		}
+	}
+
+	function getDuplicatedShowStyleVariantCount(showStyleVariant: ShowStyleVariant): number {
+		const variantsWithName = props.dragDropVariants.filter((variant: ShowStyleVariant) => {
+			const variantNameWithoutDuplicator = getNameWithRemovedDuplicateIndicator(variant)
+			return variantNameWithoutDuplicator === showStyleVariant.name && variant._id !== showStyleVariant._id
+		})
+		return variantsWithName.length
+	}
+
+	function getNameWithRemovedDuplicateIndicator(showStyleVariant: ShowStyleVariant): string {
+		return showStyleVariant.name.replace(/ +\(\d+\)$/, '')
+	}
+
+	function replaceOtherShowStyleVariant(): void {
+		const variantToReplace = props.dragDropVariants.find((variant: ShowStyleVariant) => {
+			return props.showStyleVariant.name === variant.name && props.showStyleVariant._id !== variant._id
+		})
+		if (!variantToReplace) {
+			return
+		}
+		MeteorCall.showstyles.removeShowStyleVariant(variantToReplace._id).catch(logger.warn)
+	}
+
+	function provideDuplicatedNameOptions() {
+		doModalDialog({
+			title: t('Two variants have the same name, do you want to keep both?'),
+			yes: t('Keep both'),
+			no: t('Replace'),
+			onAccept: () => {
+				addDuplicationCountToName(props.showStyleVariant)
+				removeAllQueueItems()
+			},
+			onSecondary: () => {
+				replaceOtherShowStyleVariant()
+				removeAllQueueItems()
+			},
+			onDiscard: () => {
+				addDuplicationCountToName(props.showStyleVariant)
+				removeAllQueueItems()
+			},
+			message: (
+				<React.Fragment>
```

Comment:
You are mixing abstraction levels here.
It's not too bad, just wanted to mention it :)

Your `onAccept`, `onSecondary`, and `onDiscard` is all on the same abstraction level, but `message` has gone one step "deeper". 
In general we want to strive for one abstraction level per function etc.